js/src/components/paid-ads/validateAssetGroup.test.js

```diff
@@ -120,57 +128,44 @@ describe( 'validateAssetGroup', () => {
-		it( 'When display_url_path is an empty array or an array with all empty string elements, it should pass', () => {
-			values.display_url_path = [];
+		it( `When values.${ ASSET_FORM_KEY.DISPLAY_URL_PATH } is an empty array or an array with all empty string elements, it should pass`, () => {
+			values[ ASSET_FORM_KEY.DISPLAY_URL_PATH ] = [];
 
 			expect( validateAssetGroup( values ) ).toStrictEqual( {} );
 
-			values.display_url_path = [ '', '' ];
+			values[ ASSET_FORM_KEY.DISPLAY_URL_PATH ] = [ '', '' ];
 
 			expect( validateAssetGroup( values ) ).toStrictEqual( {} );
 		} );
 
-		it( 'When display_url_path is an array where the first element is an empty string and the second is a non-empty string, it should not pass', () => {
-			values.display_url_path = [ '', 'bar' ];
+		it( `When values.${ ASSET_FORM_KEY.DISPLAY_URL_PATH } is an array where the first element is an empty string and the second is a non-empty string, it should not pass`, () => {
+			values[ ASSET_FORM_KEY.DISPLAY_URL_PATH ] = [ '', 'bar' ];
 			const error = validateAssetGroup( values );
+			const messages = error[ ASSET_FORM_KEY.DISPLAY_URL_PATH ];
 
-			expect( error ).toHaveProperty( 'display_url_path' );
-			expect( Array.isArray( error.display_url_path ) ).toBe( true );
-			expect( error.display_url_path ).toHaveLength( 1 );
-			expect( error.display_url_path ).toMatchSnapshot();
+			expect( error ).toHaveProperty( ASSET_FORM_KEY.DISPLAY_URL_PATH );
+			expect( Array.isArray( messages ) ).toBe( true );
+			expect( messages ).toHaveLength( 1 );
+			expect( messages ).toMatchSnapshot();
 		} );
 
-		it.each( ASSET_TEXT_SPECS.map( ( spec ) => [ spec.key, spec ] ) )(
-			'When the character limit is exceeded in the index %d of display_url_path, it should not pass',
-			( key, spec ) => {
-				values[ key ] = Array.from( { length: spec.min }, ( _, i ) => {
-					const limit =
-						spec.maxCharacterCounts?.[ i ] ??
-						spec.maxCharacterCounts;
-					return i.toString().repeat( limit + 1 );
-				} );
-				const error = validateAssetGroup( values );
-
-				expect( error ).toHaveProperty( key );
-				expect( Array.isArray( error[ key ] ) ).toBe( true );
-				expect( error[ key ] ).toHaveLength( spec.min );
-				expect( error[ key ] ).toMatchSnapshot();
-			}
-		);
-
-		it( 'When the character limit is exceeded in display_url_path, it should not pass', () => {
+		it( `When the character limit is exceeded in values.${ ASSET_FORM_KEY.DISPLAY_URL_PATH }, it should not pass`, () => {
 			const specs = ASSET_DISPLAY_URL_PATH_SPECS;
 			const { length } = specs;
-			values.display_url_path = Array.from( { length }, ( _, i ) => {
-				const limit = specs[ i ].maxCharacterCount;
-				return i.toString().repeat( limit + 1 );
-			} );
+			values[ ASSET_FORM_KEY.DISPLAY_URL_PATH ] = Array.from(
+				{ length },
+				( _, i ) => {
+					const limit = specs[ i ].maxCharacterCount;
+					return i.toString().repeat( limit + 1 );
+				}
+			);
```

Comment:
I was hard to understand and read the logic under this block. I would write it in a more verbose way.

Reply:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/1872/commits/80e1c00c1e98ad4c06bf3a6f0e66d2f91fe5a456.